bionic/protocols.py

```diff
@@ -197,6 +198,39 @@ def __repr__(self):
         return f"{self.__class__.__name__}(...)"
 
 
+class JsonProtocol(BaseProtocol):
+    """
+    Decorator indicating that an entity's values can be serialized to the
+    ``JSON`` format.
+    """
+
+    def get_fixed_file_extension(self):
+        return "json"
+
+    def validate(self, value):
+        if value is None:
+            return
+        if isinstance(value, list):
+            for elem in value:
+                self.validate(elem)
+            return
+        if isinstance(value, dict):
+            for key, elem in value.items():
+                self.validate(key)
```

Comment:
I think we may want to require that the key is a string, not just a valid value; it seems like JSON will accept other values but convert them to strings, which is probably not what we want.

Reply:
Ah got it. I thought keys could be ints and bools and even None but it looks like keys have to be strings. Fixed.